### PR TITLE
Add data manager to credentials generator

### PR DESCRIPTION
**Pull request by illusional**
Context: https://centrepopgen.slack.com/archives/C030X7WGFCL/p1676239341416309?thread_ts=1675743196.294449&cid=C030X7WGFCL

### CODE REVIEW

File: cpg_infra/driver.py [hunk 1]
```diff
@@ -727,6 +728,16 @@ def setup_web_access_group_memberships(self):
             member=self.analysis_group,
         )
 
+    def setup_service_account_generation_permission(self):
+        if not isinstance(self.infra, GcpInfrastructure):
+            return
+
+        self.infra.add_project_role(
```
Comment (lgruen) [inline cpg_infra/driver.py:735]: Can we add this on the service account itself instead of the whole project?

**Review by lgruen — CHANGES_REQUESTED** [review]

File: cpg_infra/driver.py (hunk 1 above)
Comment (illusional) [reply cpg_infra/driver.py]: Yep, definitely makes sense

File: cpg_infra/abstraction/base.py [hunk 2]
```diff
@@ -333,8 +345,13 @@ def create_machine_account(
...
-        self, resource_key: str, machine_account, member, project: str = None
+    def add_member_to_machine_account_role(
+        self,
+        resource_key: str,
+        machine_account,
+        member,
+        role: MachineAccountRole,
+        project: str = None,
     ) -> Any:
         print(f'Allow {member} to access {machine_account}')
```
Comment (lgruen) [inline cpg_infra/abstraction/base.py:356]: Did you mean to just have a simple `pass` here?

**Review by lgruen — APPROVED** [review]

File: cpg_infra/abstraction/base.py (hunk 2 above)
Comment (illusional) [reply cpg_infra/abstraction/base.py:356]: This is a DEV CloudInfraBase, so it's supposed to log out what it's doing.

**Pull request merged**